basculehttp/constructor.go

```diff
@@ -77,13 +76,13 @@ type constructor struct {
 	headerName          string
 	headerDelimiter     string
 	authorizations      map[bascule.Authorization]TokenFactory
-	getLogger           func(context.Context) log.Logger
+	getLogger           sallust.GetLoggerFunc
 	parseURL            ParseURL
 	onErrorResponse     OnErrorResponse
 	onErrorHTTPResponse OnErrorHTTPResponse
 }
 
-func (c *constructor) authenticationOutput(logger log.Logger, request *http.Request) (bascule.Authentication, ErrorResponseReason, error) {
+func (c *constructor) authenticationOutput(logger *zap.Logger, request *http.Request) (bascule.Authentication, ErrorResponseReason, error) {
```

Comment:
We should refactor this.  `bascule` shouldn't have a direct, compile-time dependency on any logging package.  Instead, it should allow application code to supply closures or observers for authentication events.